### PR TITLE
There should be a way to jump to an issue directly

### LINKED ISSUE #844: There is should be a way to jump to an issue directly
In the situation below, the focus is not on any element that we can interact with. The user should be able to put focus on an issue without going through the filter box first.
image

Pressing `1` should jump to first issue, `2` to the 2nd issue and so on?

### PR DESCRIPTION
**Pull request by rd1992**
Fixes #844

### CODE REVIEW

**Comment by rd1992** [comment]
@damithc okay sure. But i think we should only care about jumping to the __first__ issue. After that the user can use either T/V or the arrow keys to move through the issue list. What do you think?

**Comment by damithc** [comment]
Yes, most of the time jumping to first is enough. Jumping to n'th issue may be useful in searching. e.g. after typing a few key words, the results is narrowed to 5 issues and you notice the 4th one is the one you are looking for.

**Comment by dariusf** [comment]
Needs a rebase, and should probably take the changes in #909 into account.

@lejolly could you review?

**Comment by lejolly** [comment]
sure, @rd1992 just comment here when you're done.

**Comment by rd1992** [comment]
@lejolly done. But again i can pass this locally. Travis is failing though

**Comment by rd1992** [comment]
ran it again and it passed this time

**Comment by damithc** [comment]
Shall we make Ctrl+Down the key to jump to the issue list? It's consistent with the keyboard shortcut used to go from filter box to issue list.

**Comment by rd1992** [comment]
We could. So you want <kbd>1</kbd> , <kbd>Enter</kbd> and <kbd>Ctrl</kbd> + <kbd>Down</kbd> as the three ways of jumping to first issue? 3 different shortcuts seems quite a lot.

**Comment by damithc** [comment]
We don't need enter.
>
>

-- 
Sent from a mobile device.

File: src/main/java/ui/components/KeyboardShortcuts.java [hunk 1]
```diff
@@ -112,6 +112,7 @@
     public static final KeyCodeCombination REMOVE_FOCUS =
             new KeyCodeCombination(KeyCode.ESCAPE);
 
+    public static final Set<KeyCode> jumpToFirstIssueKeys = populateJumpToFirstIssueSet();
```
Comment (lejolly) [inline src/main/java/ui/components/KeyboardShortcuts.java]: move this higher since it belongs under the `ListPanel` part?

File: src/test/java/guitests/KeyboardShortcutsTest.java [hunk 2]
```diff
@@ -144,7 +159,9 @@ public void keyboardShortcutsTest() {
         assertEquals(5, selectedIssueId);
         
         // minimize window
-        press(KeyCode.CONTROL).press(KeyCode.N).release(KeyCode.N).release(KeyCode.CONTROL); // run this last
+        press(KeyCode.CONTROL).press(KeyCode.N).release(KeyCode.N).release(KeyCode.CONTROL); // run
```
Comment (lejolly) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: for emphasis?

File: src/test/java/guitests/KeyboardShortcutsTest.java [hunk 3]
```diff
@@ -117,22 +130,24 @@ public void keyboardShortcutsTest() {
         int issueIdExpected = issueIdBeforeMark - 1;
         push(getKeyCode("MARK_AS_READ"));
         assertEquals(issueIdExpected, selectedIssueId);
-        push(getKeyCode("UP_ISSUE")); //required since focus has changed to next issue
+        push(getKeyCode("UP_ISSUE")); // required since focus has changed to
```
Comment (lejolly) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: why are all the comments getting chopped up?

**Comment by lejolly** [comment]
ok, waiting for <kbd>Ctrl</kbd>+<kbd>Down</kbd> to be implemented.

File: src/test/java/guitests/KeyboardShortcutsTest.java (hunk 3 above)
Comment (rd1992) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: okay fixed the chopping part

File: src/test/java/guitests/KeyboardShortcutsTest.java [hunk 4]
```diff
@@ -59,7 +59,20 @@ public void keyboardShortcutsTest() {
         assertEquals(UIComponentFocusEvent.EventType.FILTER_BOX, uiComponentFocusEventType);
         clearUiComponentFocusEventType();
 
+        // jump to first issue using number key(1) or ENTER
+        push(KeyCode.ESCAPE);
+        push(KeyCode.DIGIT1);
```
Comment (lejolly) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: could we get the keycodes directly from `KeyboardShortcuts`?

File: src/test/java/guitests/KeyboardShortcutsTest.java [hunk 5]
```diff
@@ -117,7 +130,7 @@ public void keyboardShortcutsTest() {
         int issueIdExpected = issueIdBeforeMark - 1;
         push(getKeyCode("MARK_AS_READ"));
         assertEquals(issueIdExpected, selectedIssueId);
-        push(getKeyCode("UP_ISSUE")); //required since focus has changed to next issue
+        push(getKeyCode("UP_ISSUE")); // required since focus has changed to next issue
```
Comment (lejolly) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: similar to what this is doing.

File: src/test/java/guitests/KeyboardShortcutsTest.java (hunk 4 above)
Comment (rd1992) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: we could do it for single key codes. but then what do you suggest for key combinations? Like <kbd>Ctrl</kbd>+<kbd>Down</kbd>. Also since their are so many of such keys we can fix them in a separate issue no?

Comment (lejolly) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: Yeah let's just do it for single key codes first.

Comment (rd1992) [inline src/test/java/guitests/KeyboardShortcutsTest.java]: I've created a separate issue #922 to update all shortcuts here. Shouldn't be part of this PR.

**Comment by damithc** [comment]
Don't forget the documentation.

**Comment by damithc** [comment]
Our usual practice is to assign milestones to issues, not PRs.

**Comment by dariusf** [comment]
Couldn't get this to pass 4 times in a row. :fearful: Also, are reviews done? Can this be included in the release?

**Comment by lejolly** [comment]
looks fine to me, apart from it failing on travis.
